Signal/src/network/PushManager.m

```diff
@@ -231,12 +234,21 @@ - (void)application:(UIApplication *)application
         completionHandler();
     } else if ([identifier isEqualToString:PushManagerActionsShowThread]) {
         NSString *threadId = notification.userInfo[Signal_Thread_UserInfo_Key];
-        [Environment presentConversationForRecipientId:threadId];
+
+        if (threadId) {
+            [Environment presentConversationForThreadId:threadId];
+        } else {
+            OWSFail(@"%@ threadId was unexpectedly nil in action with identifier: %@", self.tag, identifier);
+        }
         completionHandler();
     } else {
         OWSFail(@"%@ Unhandled action with identifier: %@", self.tag, identifier);
         NSString *threadId = notification.userInfo[Signal_Thread_UserInfo_Key];
-        [Environment presentConversationForRecipientId:threadId];
+        if (threadId) {
+            [Environment presentConversationForThreadId:threadId];
+        } else {
+            OWSFail(@"%@ threadId was unexpectedly nil in action with identifier: %@", self.tag, identifier);
```

Comment:
Why not DRY up this nil test into `presentConversationForThreadId:`?

Reply:
A failure here likely indicates malformed notification params, which, being a K-V bag is easy to mess up, so I wanted to log the action as debug context.

Reply:
Makes sense.